content/articles/understanding-embeddings-in-machine-learning/index.md

```diff
@@ -0,0 +1,98 @@
+
+Since the technique transforms categorical variables into continuous variables, many zeros and ones will be produced. Due to this, an issue of the unmanageable number of dimensions arises, especially where the variables have a variety of unique characters. The main reason is that each item in the vector space is equally distanced and impossible to compare. In addition, there is no way to compare the categories with differences in variance when it comes to vector space. This, in turn, means it is not possible to determine whether the entities are related to each other. Extensive manual labeling that is not logical has to be done to generate one-to-one mappings and grouping, looking for similarities.
+
+In order to solve the above issue, the categorical variables must be represented well and keep the relationship information between the items. Also, categorical variables have to be minimized to keep the items of the same categories packed together. The following section explains that this can be achieved by using embeddings.
+
+### Use of embeddings in solving an encoding problem
+Embeddings represent data from the object as numbers. The vector space measures the similarities in the categories. The vectors are said to be similar if they neighbor one another. 
+
+Embeddings can be combined to work alongside other models in an online store. The models can use the same learnings between the same items instead of handling them as different separate categories. Therefore, embeddings become useful in downstreaming the models and representing the sparse data. On the downside, embeddings are complex to understand and require more computing resources compared to one-hot encodings.
+
+### Creating embeddings
+Before building embeddings, it is essential to have a supervised ML model well defined. This means that the model must be trained to transform the categories into vectors. For instance, we can assume a model that tells the type of meal the person will take based on their previous meals preferences. We can then create a model that will take our data as the input and transform it into a vector. Then the vector will predict the meal the person will take next. This implies that similar vectors refer to the meal taken frequently. The representation can be used for customization. In our case, we have used embeddings to solve a supervised problem. However, it is better to note that creating an embedding can be an unsupervised problem.
+
+### Commonly used models in embeddings
+#### PCA
+[PCA](https://www.keboola.com/blog/pca-machine-learning) method is used to reduce the variables into small subsets to minimize the size of an entity. This method makes the variables complex to understand, which may cause data loss; however, it makes the model efficient.
```

Comment:
This description of PCA is too shallow. Consider adding more information about the embedding.